aspnetcore/blazor/security/index.md

```diff
@@ -0,0 +1,482 @@
+```
+
+If `user.Identity.IsAuthenticated` is `true` and because the user is a <xref:System.Security.Claims.ClaimsPrincipal>, claims can be enumerated and membership in roles evaluated.
+
+## Authentication State Provider service
+
+Blazor server-side includes a built-in Authentication State Provider service (`AuthenticationStateProvider`) that obtains authentication state data from ASP.NET Core's server-side `HttpContext.User`. This is how authentication state integrates with existing ASP.NET Core server-side authentication mechanisms.
+
+`AuthenticationStateProvider` is the underlying service that supports `<AuthorizeView>` and `Task<AuthenticationState>`.
+
+We don't recommend using `AuthenticationStateProvider`. Use the `<AuthorizeView>` or `Task<AuthenticationState>` approaches described earlier where possible. The main drawback to using `AuthenticationStateProvider` directly is that the component isn't notified automatically if the underlying authentication state data changes.
+
+The `AuthenticationStateProvider` service can provide the current user's <xref:System.Security.Claims.ClaimsPrincipal> data, as shown in the following example:
+
+```cshtml
+@page "/"
+@inject AuthenticationStateProvider AuthenticationStateProvider
+
+<button @onclick="@LogUsername">Write user info to console</button>
+
+@code {
+    private async Task LogUsername()
+    {
+        var authState = await AuthenticationStateProvider.GetAuthenticationStateAsync();
+        var user = authState.User;
+
+        if (user.Identity.IsAuthenticated)
+        {
+            Console.WriteLine($"{user.Identity.Name} is authenticated.");
+        }
+        else
+        {
+            Console.WriteLine("The user is NOT authenticated.");
+        }
+    }
+}
+```
+
+If `user.Identity.IsAuthenticated` is `true` and because the user is a <xref:System.Security.Claims.ClaimsPrincipal>, claims can be enumerated and membership in roles evaluated.
+
+For more information on dependency injection (DI) and services, see <xref:blazor/dependency-injection> and <xref:fundamentals/dependency-injection>.
+
+## Implement a custom Authentication State Provider
+
+> [!WARNING]
+> **We don't recommend implementing a custom Authentication State Provider.**
+>
+> The built-in Authentication State Provider (`AuthenticationStateProvider`) implementation integrates with ASP.NET Core's built-in authentication mechanisms. If a custom provider is implemented, security vulnerabilities might be introduced to the app.
+
+The only common scenario where a custom Authentication State Provider (`AuthenticationStateProvider`) is implemented is in a Blazor client-side app. A custom provider can integrate with external authentication systems independently of server-side code. In Blazor client-side apps, authentication only exists to present a convenient UI to well-behaved users&mdash;authentication doesn't enforce security in Blazor client-side apps because client-side rules can be bypassed by a malicious user.
```

Comment:
```suggestion
The only common scenario where a custom Authentication State Provider (`AuthenticationStateProvider`) is implemented is in a Blazor client-side app. A custom provider can integrate with external authentication systems independently of server-side code. In Blazor client-side apps, authentication only exists to present a convenient UI to well-behaved users&mdash;authentication doesn't enforce security in Blazor client-side apps because client-side authorization rules can be bypassed by a malicious user.
```